### DISCUSSION TITLE
Using || in keyboard shortcut

### DISCUSSION THREAD

**Discussion by vid512** [bbidulock/icewm#704] [Q&A]
Due to some weird bug with FluxBox (messing up behavior of some GUI apps a bit), I'm now trying out IceWM. 

In FluxBox, I have defined a keyboard shortcut that either focuses an application if it is started already, or starts it otherwise. When I tried to do the same in IceWM, it doesn't seem to work:

~/.icewm/keys:
```
key "Super+w" wmctrl -a "Firefox" || firefox
```

This correctly switches to Firefox if it is already started, but otherwise it doesn't start Firefox. It seems like the || isn't processed in usual shell-like manner in IceWM.

Any idea how to do this simply?

**Comment by gijsbers** [comment]
What happens when you try:

```
key "Super+w" pstree
```

or:

```
key "Super+w" echo "$@" || echo false
```

**Comment by vid512** [comment]
Nothing happens, but I am probably missing some step here. IceWM is started by the display manager, and I can't see any console output from it. If it is logged somewhere, then I don't know where to look for it.

However, when I try:
```key "Super+w" firefox```
Then Win+W will correctly start another instance of Firefox.  (If that's what you wanted to test)

**Comment by gijsbers** [comment]
`lsof -p $(pgrep -x icewm) | grep -e 1w -e 2w`

gives you the file where icewm output is logged. That enables you to discover by experimenting what is the matter. I want you to become self-sufficient, because I don't wish to answer loads of questions by every new user, as you may understand, hopefully.

**Comment by vid512** [comment]
Both your help and your approach are appreciated. I prefer to understand things I use.

So, using
``` key "Super+w" echo "$@" || echo false ```
outputs
``` $@ || echo false ```

That means IceWM directly executes echo command, with rest as arguments, not passing it through bash, which would process the || chaining. 

So, the solution is:
``` key "Super+w" bash -c "wmctrl -a \"Firefox\" || firefox" ```

So far it seems to work as expected. Thanks.

**Comment by gijsbers** [comment]
I'll give you a more icewm-ish style solution as a reward for your hard work:

```
icesh if -c firefox -l then activate else runonce firefox
```

Depending on your configuration and goals this may also work:

```
icesh -c firefox runonce firefox
```
